### PR TITLE
Add storage provider configuration

### PR DESCRIPTION
**Pull request by zapplebee**
## About the changes

Previously, there was no way to set the storage provider for the underlying Node SDK. This change exposes that option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +125,7 @@ Keep-Alive: timeout=5
...
 | clientKeysHeaderName      | `CLIENT_KEY_HEADER_NAME`         | "authorization" |    no    | The name of the HTTP header to use for client keys. Incoming requests must set the value of this header to one of the Proxy's `clientKeys` to be authorized successfully.                                                                                                                                                                                                                                                                                                |
+| storageProvider           | n/a                              | `undefined`     |    no    | Register a custom storageProvider.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (thomasheartman) [inline README.md:128]:
```suggestion
| storageProvider           | n/a                              | `undefined`     |    no    | Register a custom storage provider. Refer to the [section on custom storage providers in the Node.js SDK's readme](https://github.com/Unleash/unleash-client-node/#custom-store-provider) for more information.                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Hey, @zapplebee! Thanks for the PR; this seems like a very reasonable request and the implementation looks sound 🙌🏼  My only request is that we add a bit more information about storage providers to the table. I've added a suggestion that you can use, or you can provide your own if you'd prefer 😄

**Comment by thomasheartman** [comment]
Oh, I see the lint check failed due to some code style issues and unused variables. Would you mind cleaning that up? I'll have a proper look again next week; it's getting a bit late here now ☺️

**Comment by zapplebee** [comment]
Yeah, no worries.

I'll get that in today.

**Pull request merged**